src/aggregate_utils.py

```diff
@@ -24,22 +46,45 @@ def download_all_promos(verbose=True):
     all_promos = []
 
     for cursor in range(0, total_num_promos, step):
-        store_data = download_store_data(cursor=cursor, step=step, verbose=verbose)
+        store_data = download_store_data(
+            cursor=cursor, step=step, verbose=verbose)
         promos = extract_upcoming_promotions(store_data)
 
         all_promos += promos
 
     return all_promos
 
 
+def find_title(promotions, title):
+    """
+    Search for title in promotions.
+
+    args: promotions: List[dict], title: str
+
+    Returns: First potential match.
+    """
+    title = title.lower()
+    for e in promotions:
+        if title in e["title"].lower():
+            print(e)
+            return e
+
+
+def find_title_regex(promotions, title):
+    """
+    Regex search for title in promotions.
+
+    args: promotions: List[dict], title: str
+
+    Returns: All potential matches.
+    """
+    return [e for e in promotions if title in re.search(title, e["title"], re.IGNORECASE)]
+
+
```

Comment:
- I would include the search functions in a separate file, e.g. `src/search_utils.py` due to `import re`.
- I would add a few unit tests for the new functions.

Reply:
Feel free to add this func if you think it may be useful! :)

Reply:
Thanks. I will do! 👍